client/src/components/dataset/sidebars/GenerateGeopoints.jsx

```diff
@@ -0,0 +1,136 @@
+  const translationId = `select_column_${latOrLong}`;
+  return (
+    <div className="inputGroup">
+      <label htmlFor={columnName}>
+        <FormattedMessage id={translationId} />
+      </label>
+      <SelectMenu
+        name={columnName}
+        value={value}
+        onChange={onChange}
+        options={numberColumnOptions(columns)}
+      />
+    </div>
+  );
+}
+
+SelectColumn.propTypes = {
+  columns: PropTypes.object.isRequired,
+  latOrLong: PropTypes.string.isRequired,
+  onChange: PropTypes.func.isRequired,
+  value: PropTypes.string,
+};
+
+export default class GenerateGeopoints extends Component {
+  constructor() {
+    super();
+    this.state = {
+      transformation: Immutable.fromJS({
+        op: 'core/generate-geopoints',
+        args: {
+          columnNameLat: null,
+          columnNameLong: null,
+          ColumnTitleGeo: '',
+        },
+        onError: 'fail',
+      }),
+    };
+    this.handleSelectColumn = this.handleSelectColumn.bind(this);
+    this.isValidTransformation = this.isValidTransformation.bind(this);
+    this.handleChangeColumnTitleGeo = this.handleChangeColumnTitleGeo.bind(this);
+  }
+
+  handleSelectColumn(value, latOrLong) {
+    const { transformation } = this.state;
+    const columnName = 'columnName'.concat(_.capitalize(latOrLong));
+    this.setState({
+      transformation: transformation.setIn(['args', columnName], value),
+    });
+  }
+
+  isValidTransformation() {
+    const { transformation } = this.state;
+    return transformation.getIn(['args', 'columnNameLat']) != null
+      && transformation.getIn(['args', 'columnNameLong']) != null
+      && transformation.getIn(['args', 'columnTitleGeo']) !== '';
+  }
+
+  handleChangeColumnTitleGeo(value) {
+    const { transformation } = this.state;
+    this.setState({
+      transformation: transformation.setIn(['args', 'columnTitleGeo'], value),
+    });
+  }
+
+  render() {
+    const { onClose, onApply, columns } = this.props;
+    const args = this.state.transformation.get('args');
+    return (
+      <div
+        className="DataTableSidebar"
+      >
+        <SidebarHeader onClose={onClose}>
+          <FormattedMessage id="generate_geopoints" />
+        </SidebarHeader>
+        <div className="inputs">
+          <SelectColumn
+            columns={columns}
+            latOrLong="lat"
+            onChange={value => this.handleSelectColumn(value, 'lat')}
+            value={args.get('columnNameLat')}
+          />
+          <SelectColumn
+            columns={columns}
+            latOrLong="long"
+            onChange={value => this.handleSelectColumn(value, 'long')}
+            value={args.get('columnNameLong')}
+          />
+          <div className="inputGroup">
+            <label
+              htmlFor="titleTextInput"
+            >
+              <FormattedMessage id="column_title_geo" />
+            </label>
+            <input
+              value={args.get('columnTitleGeo')}
```

Comment:
This generates a console error on first input - `Warning: GenerateGeopoints is changing an uncontrolled input of type text to be controlled.`

I guess that's because the value is initially null. Perhaps we should change to something like `value={args.get('columnTitleGeo') || ''}`

Reply:
:+1: